src-built-in/components/UIAPI/reducers/linker.spec.ts

```diff
@@ -0,0 +1,168 @@
+        const action = {
+            type: 'some invalid type',
+            payload: {}
+        };
+        const output = reducer(undefined, action);
+        assert.deepEqual(output, initialState);
+    });
+    
+    // Init request shouldn't change any state value
+    it('Should handle init', () => {
+        const output: any = reducer(initialState, actions.init());
+        const expectedOutput = loop(initialState, Cmd.run(initializeLinker, {
+                successActionCreator: initSuccess,
+                args: [initialState]
+            }))
+        assert.deepEqual(expectedOutput, output);
+    });
+
+    it('Should handle init success', () => {
+        const payloadValue = {
+            channels: {
+                20: channel(20, false, 'green'),
+                30: channel(30, false, 'red'),
+            },
+            nameToId: {
+                'green': 20,
+                'red': 30
+            },
+            isAccessibleLinker: true,
+            windowIdentifier: {
+                windowName: "welcome component"
+            },
+            actives: 0,
+            processingRequest: false
+        };
+        const initSuccessAction = actions.initSuccess(payloadValue);
+        const output: any = reducer(initialState, initSuccessAction);
+        const expectedOutput = loop(payloadValue, Cmd.run(fitDOM));
+        assert.deepEqual(expectedOutput, output);
+    });
+
+    // Cleaning up shouldn't change the state
+    it('Should handle linker cleanup', () => {
+        const output: any = reducer(initialState, actions.cleanUp());
+        const expectedOutput = loop(initialState, Cmd.run(cleanUp));
+        assert.deepEqual(expectedOutput, output);
+    });
+
+    // Toggle channel request set the "prosessingRequest" property on the linker state to "true"
+    it('Should handle toggle channel request', () => {
+        const expectedInput = {
+            channels: {
+                2: {
+                    name: "hi",
+                    color: "green",
+                    active: true,
+                    id: 2,
+                    border: "green"
+                }
+            },
+            nameToId: {
+                "hi": 2
+            },
+            isAccessibleLinker: true,
+            windowIdentifier: {},
+            processingRequest: false,
+            actives: 1
+        }
+        const changedState = {
+            ...expectedInput,
+            processingRequest: true
+        };
+        changedState.processingRequest = true;
+        const output: any = reducer(expectedInput, actions.toggleChannel(2));
+        const cmd = Cmd.run(linkChannel, {
+            successActionCreator: () => toggleSuccess(2),
+            failActionCreator: () => toggleFailure(),
+            args: ["hi", true, {}, 1]
+        });
+        const expectedOutput = loop(changedState, cmd);
+        assert.deepEqual(JSON.stringify(expectedOutput), JSON.stringify(output));
```

Comment:
You're using `deepEqual` for string comparison, whats the point?

Reply:
Yeah you're right I can use `equal` here